internal/assets/operations.go

```diff
@@ -99,49 +99,49 @@ func (am *assetManager) PrepareOperation(ctx context.Context, op *core.Operation
 	}
 }
 
-func (am *assetManager) RunOperation(ctx context.Context, op *core.PreparedOperation) (outputs fftypes.JSONObject, complete bool, err error) {
+func (am *assetManager) RunOperation(ctx context.Context, op *core.PreparedOperation) (outputs fftypes.JSONObject, phase core.OpPhase, err error) {
 	switch data := op.Data.(type) {
 	case createPoolData:
 		plugin, err := am.selectTokenPlugin(ctx, data.Pool.Connector)
 		if err != nil {
-			return nil, false, err
+			return nil, core.OpPhaseInitializing, err
 		}
-		complete, err = plugin.CreateTokenPool(ctx, op.NamespacedIDString(), data.Pool)
-		return nil, complete, err
+		phase, err = plugin.CreateTokenPool(ctx, op.NamespacedIDString(), data.Pool)
+		return nil, phase, err
 
 	case activatePoolData:
 		plugin, err := am.selectTokenPlugin(ctx, data.Pool.Connector)
 		if err != nil {
-			return nil, false, err
+			return nil, core.OpPhaseInitializing, err
 		}
-		complete, err = plugin.ActivateTokenPool(ctx, data.Pool)
-		return nil, complete, err
+		phase, err = plugin.ActivateTokenPool(ctx, data.Pool)
+		return nil, phase, err
 
 	case transferData:
 		plugin, err := am.selectTokenPlugin(ctx, data.Pool.Connector)
 		if err != nil {
-			return nil, false, err
+			return nil, core.OpPhaseInitializing, err
 		}
 		switch data.Transfer.Type {
 		case core.TokenTransferTypeMint:
-			return nil, false, plugin.MintTokens(ctx, op.NamespacedIDString(), data.Pool.Locator, data.Transfer, data.Pool.Methods)
+			return nil, core.OpPhaseInitializing, plugin.MintTokens(ctx, op.NamespacedIDString(), data.Pool.Locator, data.Transfer, data.Pool.Methods)
```

Comment:
It's the _phase_ we're in as a result of the error in this layer of the code, rather than the operation state.

e.g. this error occurred before we submitted the operation to the connector, so the calling code can decide if that means we don't store the operation at all, or we store it in state `Initializing`

Reply:
I have not analyzed very scenario to know if the error path is such, that it could *never succeed*.
If I had done that (quite large) piece of work everywhere, I could have introduced a phase of `OpPhaseUnprocessable` to mean - "this is never going to work, tell them it's broken"

Reply:
I noticed in some managers you used `operations.ErrTernary`, but not here.

Reply:
Sorry - yes, this should be `operations.ErrTernary` almost certainly - I'd thought this was an error return.

Reply:
ok - added a commit for that